skins/cat17/src/scripts/membershipForm.js

```diff
@@ -187,21 +187,30 @@ $( function () {
 			),
 			stateKey: [
 				'membershipFormContent.membershipType',
-				function ( store ) {
-					return {
-						isValid: store.membershipFormContent.membershipType !== null,
-						dataEntered: store.membershipFormContent.membershipType !== null
-					}
-				}
+				WMDE.StateAggregation.Membership.membershipTypeIsValid
+			]
+		},
+		{
+			viewHandler: WMDE.View.SectionInfo.createMembershipTypeSectionInfo(
+				$( '.state-bar .member-type' ),
+				{
+					'sustaining': 'icon-favorite',
+					'active': 'icon-flash_on'
+				},
+				WMDE.FormDataExtractor.mapFromRadioLabelsShort( $( '#type-membership .wrap-input' ) ),
+				{ 'sustaining': '', 'active': '' }
+			),
+			stateKey: [
+				'membershipFormContent.membershipType',
+				WMDE.StateAggregation.Membership.membershipTypeIsValid
 			]
 		},
 		{
 			viewHandler: WMDE.View.SectionInfo.createDonorTypeSectionInfo(
-				$( '.donor-type' ),
+				$( '.state-bar-lateral .donor-type, .state-bar-detailed .donor-type' ),
 				{
 					'person': 'icon-account_circle',
-					'firma': 'icon-work',
-					'anonym': 'icon-visibility_off'
```

Comment:
For the record: this is a desired change, `anonym` is not an option for memberships.